go/pkg/tree/tree.go

```diff
@@ -27,8 +27,12 @@ type treeNode struct {
 }
 
 type fileNode struct {
-	Chunker              *chunker.Chunker
-	IsExecutable         bool
+	Chunker      *chunker.Chunker
+	IsExecutable bool
+}
+
+type fileSysNode struct {
```

Comment:
why do you need to have this separately?

Reply:
I'll later add `symlinkNode` into it. `fileSysNode` is a union representing a regular file, and empty directory or a symlink. (It's really similar to `treeNode`, except that it only contains a single item)

Reply:
I see.